tools/clang_format_wrapper.py

```diff
@@ -0,0 +1,136 @@
+#!/usr/bin/env python
+"""Wrapper script for `git-clang-format` to avoid race conditions when
+merging/rebasing CPP files with merge conflicts."""
+from collections import namedtuple
+from os import environ as env
+from subprocess import check_output
+from typing import List, Iterable
+import argparse
+import enum
+import logging
+import sys
+
+logger = logging.getLogger(__name__)
+
+
+class GitState(enum.Enum):
+    """
+    Excerpt from https://git-scm.com/docs/git-status
+
+    For paths with merge conflicts, X and Y show the modification states
+    of each side of the merge. For paths that do not have merge conflicts,
+    X shows the status of the index, and Y shows the status of the work
+    tree. For untracked paths, XY are ??. Other status codes can be
+    interpreted as follows:
+    M = modified
+    A = added
+    D = deleted
+    R = renamed
+    C = copied
+    U = updated but unmerged
+    """
+
+    ADDED = "A"
+    COPIED = "C"
+    DELETED = "D"
+    MODIFIED = "M"
+    RENAMED = "R"
+    UNMODIFIED = " "
+    UNTRACKED = "?"
+    UPDATED_BUT_UNMERGED = "U"
+
+
+GitEdit = namedtuple("GitEdit", ["x", "y", "path"])
+
+
+def added_and_modified_paths() -> Iterable[str]:
+    """Inspect `git status` output.
+
+    Return paths to be modified by `git-clang-format`.
+    """
+
+    command = ["git", "status", "--porcelain", "--untracked-files=no"]
+    logger.debug("Executing `%s`", " ".join(command))
+    git_status_output = check_output(args=command, text=True)
+
+    edits = [
+        edit for edit in git_status_output.rstrip().split("\n") if edit != ""
+    ]
+    logger.debug(
+        "Found %d path(s) with state change in GIT status output", len(edits)
+    )
+
+    if edits:
+        logger.debug("Pending GIT edits:\n%s", "\n".join(edits))
+
+    return (
+        git_edit.path
+        for git_edit in (
+            GitEdit(edit[0:1], edit[1:2], edit[3:]) for edit in edits
+        )
+        if git_edit.x in (GitState.MODIFIED.value, GitState.ADDED.value)
+    )
```

Comment:
I'd rather return a sequence of `GitEdits` instead of just the paths and  filter on the consumer side (and name the function `get_git_status`), but no need to change it now.